pkg/utils/common.go

```diff
@@ -130,3 +135,101 @@ func MigrateToNewDirStructure(orig string) string {
 		return filepath.Join(git.WegoRoot, git.WegoAppDir, f[len(f)-2], f[len(f)-1])
 	}
 }
+
+type ConfigStatus int
+
+const (
+	Missing ConfigStatus = iota
+	Partial
+	Embedded
+	Valid
+)
+
+func (cs ConfigStatus) String() string {
+	switch cs {
+	case Missing:
+		return "Missing"
+	case Partial:
+		return "Partial"
+	case Embedded:
+		return "Embedded"
+	case Valid:
+		return "Valid"
+	default:
+		return "UnknownStatus"
+	}
+}
+
+type WalkResult struct {
+	Status ConfigStatus
+	Path   string
+}
+
+func (wr WalkResult) Error() string {
+	return fmt.Sprintf("found %s: with status: %s", wr.Path, wr.Status)
+}
+
+func FindCoreConfig(dir string) WalkResult {
+	err := filepath.WalkDir(dir,
+		func(path string, _ fs.DirEntry, err error) error {
+			if err != nil {
+				return nil
+			}
+
+			if !strings.HasSuffix(path, ".yml") && !strings.HasSuffix(path, ".yaml") {
+				return nil
+			}
+
+			data, err := ioutil.ReadFile(path)
+			if err != nil {
+				return nil
+			}
+
+			r := bytes.NewReader(data)
+			decoder := yaml.NewDecoder(r)
+			docs := []map[string]interface{}{}
+
+			for {
+				var entry map[string]interface{}
+				if err := decoder.Decode(&entry); err == io.EOF {
+					break
+				}
+
+				docs = append(docs, entry)
+			}
+
+			rootNode := dasel.New(docs)
+			foundPartial := false
+
+			_, err = rootNode.QueryMultiple(".(kind=HelmRelease)(.metadata.name=ww-gitops)")
+			if err == nil {
+				foundPartial = true
+			}
+
+			_, err = rootNode.QueryMultiple(".(kind=GitRepository)(.metadata.name=ww-gitops)")
+			if err != nil {
+				if foundPartial {
+					return WalkResult{Status: Partial, Path: path}
+				}
+
+				return nil
+			}
+
+			val, err := rootNode.Query(".[#]")
+			if err != nil {
+				return nil
+			}
+
+			if val.InterfaceValue() != 2 { // we have two manifests
```

Comment:
This also feels pretty fragile, I wouldn't be surprised if we end up with more manifests.

Reply:
That seems quite plausible (and "2" should be a constant) but I don't think it's reasonable to assume that when we add manifests to core we won't need to do any updates to the upgrade process unless we're prepared to lock down the directory structure a bit.